crates/hir/src/code_model.rs

```diff
@@ -1276,6 +1276,18 @@ impl TypeParam {
         }
     }
 
+    pub fn trait_bounds(self, db: &dyn HirDatabase) -> Vec<Trait> {
```

Comment:
The return type here is wrong (it loses info about type params). We might want to introduce TraitRef to code model.